srml/support/procedural/src/storage/transfo.rs

```diff
@@ -0,0 +1,619 @@
+			name,
+			getter,
+			storage_type,
+			..
+		} = sline;
+
+		if let Some(getter) = getter {
+			let get_fn = &getter.getfn.content;
+			let (is_simple, extracted_opt, stk, gettype) = match storage_type {
+				DeclStorageType::Simple(ref st) => (true, ext::extract_type_option(st), None, st),
+				DeclStorageType::Map(ref map) => (false, ext::extract_type_option(&map.value), Some(&map.key), &map.value),
+			};
+
+			let typ = extracted_opt.unwrap_or(quote!(#gettype));
+			let item = if is_simple {
+				quote!{
+					pub fn #get_fn() -> #gettype {
+						<#name<#traitinstance> as #scrate::storage::generator::StorageValue<#typ>> :: get(&#scrate::storage::RuntimeStorage)
+					}
+				}
+			} else {
+				let kty = stk.expect("is not simple; qed");
+				// map
+				quote!{
+					pub fn #get_fn<K: #scrate::storage::generator::Borrow<#kty>>(key: K) -> #gettype {
+						<#name<#traitinstance> as #scrate::storage::generator::StorageMap<#kty, #typ>> :: get(key.borrow(), &#scrate::storage::RuntimeStorage)
+					}
+				}
+			};
+			items.push(item);
+		}
+	}
+
+	quote!( #( #items )* )
+}
+
+fn store_functions_to_metadata (
+	scrate: &proc_macro2::TokenStream,
+	storage_lines: &ext::Punctuated<DeclStorageLine, Token![;]>,
+) -> proc_macro2::TokenStream {
+
+	let mut items = Vec::new();
+	for sline in storage_lines.inner.iter() {
+		let DeclStorageLine {
+			attrs,
+			name,
+			storage_type,
+			..
+		} = sline;
+
+		let (is_simple, extracted_opt, stk, gettype) = match storage_type {
+			DeclStorageType::Simple(ref st) => (true, ext::extract_type_option(st), None, st),
+			DeclStorageType::Map(ref map) => (false, ext::extract_type_option(&map.value), Some(&map.key), &map.value),
+		};
+
+		let is_option = extracted_opt.is_some();
+		let typ = extracted_opt.unwrap_or(quote!( #gettype ));
+		let stype = if is_simple {
+			let styp = typ.to_string().replace(" ","");
+			quote!{
+				#scrate::storage::generator::StorageFunctionType::Plain(
+					#scrate::storage::generator::DecodeDifferent::Encode(#styp),
+				)
+			}
+		} else {
+			let kty = stk.expect("is not simple; qed");
+			let kty = quote!(#kty).to_string().replace(" ","");
+			let styp = typ.to_string().replace(" ","");
+			quote!{
+				#scrate::storage::generator::StorageFunctionType::Map {
+					key: #scrate::storage::generator::DecodeDifferent::Encode(#kty),
+					value: #scrate::storage::generator::DecodeDifferent::Encode(#styp),
+				}
+			}
+		};
+		let modifier = if !is_option {
+			quote!{
+				#scrate::storage::generator::StorageFunctionModifier::Default
+			}
+		} else {
+			quote!{
+				#scrate::storage::generator::StorageFunctionModifier::Optional
+			}
+		};
+		let mut docs = Vec::new();
+		for attr in attrs.inner.iter() {
+			if let Some(syn::Meta::NameValue(syn::MetaNameValue{
+				ref ident,
+				ref lit,
+				..
+			})) = attr.interpret_meta() {
+				if ident == "doc" {
+					docs.push(quote!(#lit));
+				}
+			}
+		}
+		let str_name = quote!(#name).to_string().replace(" ","");
```

Comment:
`name` is already an identifier and this already has the function `to_string()` and why do you need to replace whitespaces? Rust identifier can not contain any whitespace.

Reply:
This is one of the point of attention I mentioned for this PR : the result of syn::Type stringified in a quote expression is filled with spaces.
for instance 'T::Key' will stringify to '"T :: Key"'.
```
u32                                                                                                                                                                                                                                          
u32                                                                                                                                                                                                                                          
T :: Hash                                                                                                                                                                                                                                    
T :: BlockNumber                                                                                                                                                                                                                             
T :: Hash                                                                                                                                                                                                                                    
T :: Hash                                                                                                                                                                                                                                    
T :: Digest                                                                                                                                                                                                                                  
Vec < EventRecord < T :: Event > >                                                                                                                                                                                                           
u32                                                                                                                                                                                                                                          
T :: Hash                                                                                                                                                                                                                                    
T :: BlockNumber                                                                                                                                                                                                                             
T :: Hash                                                                                                                                                                                                                                    
T :: Hash                                                                                                                                                                                                                                    
T :: Digest  
```
I remember seing some issue related to this when checking.
Also, thanks for all the suggestion, I will be addressing them start of next week.